gems/sorbet-runtime/lib/types/private/methods/decl_builder.rb

```diff
@@ -115,7 +124,7 @@ def on_failure(*args)
       self
     end
 
-    def abstract
+    def abstract(&blk)
       check_live!
 
       case decl.mode
```

Comment:
It looks like sorbet-runtime already implements the correct behavior as a result of the chosen implementation, but can we add tests that ensure that the "cannot be repeated in a single signature" errors are correctly reported when repeating `abstract` (and all other builders) once outside and once inside the block?

For example

```ruby
# typed: strict

require_relative './lib/sorbet-runtime'

class Parent
  extend T::Sig
  extend T::Helpers

  abstract!

  sig.abstract {abstract.void}
  def foo; end
end

class Child < Parent
  sig.override {void}
  def foo
    puts 'Child#foo'
  end
end

Child.new.foo
```

(this correctly raises right now, but there are no tests for this)